### DISCUSSION TITLE
Native multilingualism in Gutenberg: Future development of synced patterns / synced pattern overrides

### DISCUSSION THREAD

**Discussion by Faton1995** [WordPress/gutenberg#63805] [Ideas]
Hello

I have been thinking about how to further develop the technology behind "Synced Pattern Overrides" and use it for native multilingualism, which will be introduced with the fourth Gutenberg phase. Perhaps a developer who is already working on / researching native multilingualism can take a look at this.

My thoughts on this:

1. Translating blocks:

Reuse / further develop the technique of Synced Pattern Overrides so that blocks can be translated into different languages. The system should be designed in a way that it can be expanded, e.g., plugins: DeepL with manual post-processing.

2. Displaying blocks in all languages:

Reuse / further develop the technique of Synced Pattern. This gives us the possibility to display a block in all languages. This is useful for language-neutral elements such as logos, buttons, or certain graphics.

3. How can I link already translated blocks?

I don't have a technical idea on how to approach this. But it should work in this direction: https://wpml.org/faq/how-to-link-already-translated-pages/

4. Hide blocks in specific languages:
Be able to completely hide blocks in specific languages.

**Comment by bacoords** [comment]
You can probably create a multilingual-like experience using synced patterns, but I don't think it will be a maintainable path forward, more like a workaround.

The main issue (and I'm struggling with this for synced patterns as well) is that it's not clear where/how synced patterns manage their unique data. It seems to be stored inside the block attributes. If synced patterns could store their data as postmeta (i.e. outside of the content area), then we could start to consider doing more creative things with them and their data, like having translations managed.

But I think with multilingual the work will have to start data-first (where does multilingual content/data live in the database) before the core team even starts to think about what the user experience of that would be like.
